msp/mspimpl.go

```diff
@@ -890,9 +890,13 @@ func (msp *bccspmsp) getCertificationChainIdentifierFromChain(chain []*x509.Cert
 // is regenerated to have a Low-S signature.
 func (msp *bccspmsp) sanitizeCert(cert *x509.Certificate) (*x509.Certificate, error) {
 	if isECDSASignedCert(cert) {
+		validityOpts := msp.getValidityOptsForCert(cert)
+		if bytes.Equal(cert.RawSubject, cert.RawIssuer) && cert.IsCA {
```

Comment:
A couple of points:

1. why isn't the `cert.IsCa` check sufficient?
2. could we please add some comments here? Linking the issue with the explanation
3. can you initialise `parentCert` to `cert`, negate the condition of the if so that we end up calling `sanitizeECDSASignedCert` only once?
4. iiuc, `if cert.IsCA && len(chain) == 1 {` will never be hit again as the code is structured today - is that correct? If so I'd remove this

Reply:
> A couple of points:
> 
>     1. why isn't the `cert.IsCa` check sufficient?

There is is actually described in my previous answer to @adecaro, in short I need to identify the attempt to sanitize root CA certificates only, cause after the changes introduced in v1.19 they moved away from bitwise comparison. Therefore it fails to build the chain for root CA, but there is no need to build the chain by calling cert.Verify in that particular case, since it's self signed cert. But you are right IMO the better way would be to change it to the `cert.CheckSignatureFrom(cert) == nil`. I will fix and push the code a little bit later, would like to check it first.

> 
>     2. could we please add some comments here? Linking the issue with the explanation

Good point, will do it.

> 
>     3. can you initialise `parentCert` to `cert`, negate the condition of the if so that we end up calling `sanitizeECDSASignedCert` only once?

Will do.

> 
>     4. iiuc, `if cert.IsCA && len(chain) == 1 {` will never be hit again as the code is structured today - is that correct? If so I'd remove this

This still covers intermediate certificates.